share/completions/firefox.fish

```diff
@@ -0,0 +1,42 @@
+# X11 options
+complete -c firefox -l display -r -d "X display to use"
+complete -c firefox -l sync -d "Make X calls synchronous"
+complete -c firefox -l g-fatal-warnings -d "Make all warnings fatal"
+
+# Firefox options
+complete -c firefox -s h -l help -d "Print this message"
+complete -c firefox -s v -l version -d "Print Firefox version"
+complete -c firefox -l full-version -d "Print Firefox version, build and platform build ids"
+complete -c firefox -s P -r -d "Start with <profile>"
```

Comment:
Nice work, thanks! Some minor comments below:

I think we can even suggest possible profiles:

```suggestion
complete -c firefox -s P -r -d "Start this profile" -fa '(string replace -rf "Name=(.*)" \'$1\' < ~/.mozilla/firefox/profiles.ini)'
```